src/sparseml/pytorch/sparsification/pruning/modifier_pruning_acdc.py

```diff
@@ -0,0 +1,261 @@
+    MagnitudePruningParamsScorer,
+)
+from sparseml.pytorch.sparsification.pruning.scorer import PruningParamsScorer
+
+
+__all__ = ["ACDCPruningModifier"]
+
+
+@PyTorchModifierYAML()
+class ACDCPruningModifier(BasePruningModifier):
+    """
+    Implementation of
+    Alternating Compressed/DeCompressed Training of Deep Neural Networks:
+    https://arxiv.org/pdf/2106.12379.pdf.
+    AC/DC performs co-training of sparse and dense models, and can return both an
+    accurate sparse model, and a dense model.
+    | Sample yaml:
+    |   !ACDCPruningModifier
+    |       compression_sparsity: 0.9
+    |       start_epoch: 0
+    |       end_epoch: 100
+    |       update_frequency: 5
+    |       params: __ALL_PRUNABLE__
+    |       global_sparsity: True
+
+    :param compression_sparsity: The sparsity enforced during the compression phase.
+    :param start_epoch: The epoch to start the modifier at
+    :param end_epoch: The epoch to end the modifier at
+    :param update_frequency: The length (in epochs) of compression/decompression phase
+    :param params: A list of full parameter names or regex patterns of names to apply
+        pruning to.  Regex patterns must be specified with the prefix 're:'. __ALL__
+        will match to all parameters. __ALL_PRUNABLE__ will match to all ConvNd
+        and Linear layers' weights. If a sparsity to param mapping is defined by
+        final_sparsity, then params should be set to []
+    :param global_sparsity: set True to enable global pruning. if False, pruning will
+        be layer-wise. Default is True
+    :param leave_enabled: True to continue masking the weights after end_epoch,
+        False to stop masking. Should be set to False if exporting the result
+        immediately after or doing some other prune. Default is True
+    :param mask_type: String to define type of sparsity (options: ['unstructured',
+        'channel', 'filter']), List to define block shape of a parameters in and out
+         channels, or a SparsityMaskCreator object. default is 'unstructured'
+    :param momentum_buffer_reset: set True to reset momentum buffer
+        before algorithm enters a consecutive decompression phase.
+        According to the paper:
+            "once all weights are re-introduced, it is beneficial
+            to reset to 0 the gradient momentum term of the optimizer;
+            this is particularly useful for the weights
+            that were previously pruned, which would otherwise
+            have stale versions of gradients."
+        Default is True
+    """
+
+    def __init__(
+        self,
+        compression_sparsity: float,
+        start_epoch: float,
+        end_epoch: float,
+        update_frequency: float,
+        params: Union[str, List[str]],
+        global_sparsity: bool = True,
+        leave_enabled: bool = True,
+        momentum_buffer_reset: bool = True,
+        mask_type: Union[str, List[int], PruningMaskCreator] = "unstructured",
```

Comment:
```suggestion
        mask_type: str = "unstructured",
```

only supports strings now